library/core/src/fmt/num.rs

```diff
@@ -398,7 +424,7 @@ macro_rules! impl_Exp {
                 exponent += 2;
             }
             // n is <= 99, so at most 2 chars long
-            let mut n = n as isize; // possibly reduce 64bit math
+            let mut n = n as i8; // possibly reduce 64bit math
```

Comment:
The reason this casts to `isize` is that operations at the "native integer size" for the hardware are likely to be faster than larger **or** smaller integers.